### PR TITLE
SAML ID provider cleanup

### PR DESCRIPTION
**Pull request by mfrances17**
## Motivation
Several fields in the original SAML identity provider were not working in the first pass, to be fixed later. 

## Brief Description
This fixes it so all SAML fields should be working as expected. Any fields that are not working from here on should be filed as issues.

## Verification Steps
Verify that the following fields now work as expected in the SAML ID provider config, as they worked in the old console (set the value in the old console and verify that the new console reads the value, etc):
Single logout service URL
NameID policy format
Principal type
Principal attribute
Want AuthnRequests signed
Signature algorithm
SAML signature key name
Validating X509 certificates
Pass subject
Allowed clock skew
Comparison
AuthnContext ClassRefs
AuthnContext DeclRefs
Stored tokens readable
Hide on login page

## Checklist:
- [x] Code has been tested locally by PR requester
- [x] User-visible strings are using the react-i18next framework (useTranslation)
- [x] Help has been implemented

## Additional Notes
Cypress tests coming in separate PR.

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -5,7 +5,6 @@
   "requires": true,
   "packages": {
     "": {
-      "name": "keycloak-admin-ui",
```
Comment (jonkoops) [inline package-lock.json:8]: This change is generated by older versions of NPM, but makes no functional difference. Make sure to update to the latest version of NPM and this change should disappear. You can upgrade NPM by running `npm install -g npm@latest`. 

Alternatively using the latest version of Node should also resolve this, if you have NVM tyou can run `nvm install 16 && nvm alias default 16`.

File: src/identity-providers/add/AdvancedSettings.tsx [hunk 2]
```diff
@@ -30,7 +30,7 @@ const LoginFlow = ({
   useFetch(
     () => adminClient.authenticationManagement.getFlows(),
     (flows) =>
-      setFlows(flows?.filter((flow) => flow.providerId === "basic-flow")),
+      setFlows(flows.filter((flow) => flow.providerId === "basic-flow")),
```
Comment (jonkoops) [inline src/identity-providers/add/AdvancedSettings.tsx:33]: Nice, thanks for cleaning that up!

File: src/identity-providers/add/DescriptorSettings.tsx [hunk 3]
```diff
@@ -45,6 +45,12 @@ const Fields = ({ readOnly }: DescriptorSettingsProps) => {
     name: "config.validateSignature",
   });
 
+  const principalType = useWatch({
+    control: control,
```
Comment (jonkoops) [inline src/identity-providers/add/DescriptorSettings.tsx:49]: If the value has the same name as the key the part after the colon can be omitted.
```suggestion
    control,
```

File: src/identity-providers/add/ReqAuthnConstraintsSettings.tsx [hunk 4]
```diff
@@ -31,23 +31,23 @@ export const ReqAuthnConstraints = () => {
...
+          name="config.authnContextComparisonType"
           defaultValue={comparisonValues[0]}
           control={control}
           render={({ onChange, value }) => (
             <Select
               toggleId="comparison"
               required
               direction="up"
-              onToggle={() => setSyncModeOpen(!syncModeOpen)}
+              onToggle={() => setComparisonOpen(!comparisonOpen)}
```
Comment (jonkoops) [inline src/identity-providers/add/ReqAuthnConstraintsSettings.tsx:42]: The `onToggle` prop provides a value in the callback to set the open state so we don't have to control this ourselves.
```suggestion
              onToggle={(isExpanded) => setComparisonOpen(isExpanded)}
```

**Review by jonkoops — CHANGES_REQUESTED** [review]

File: package-lock.json (hunk 1 above)
Comment (mfrances17) [reply package-lock.json:8]: hmmm, i'm already using v16.1.0 of node i'm surprised that's the issue, but i will try updating, thanks

Comment (mfrances17) [reply package-lock.json]: i don't know why i bother questioning it, you were right - fixed when i updated to the latest npm, thanks!

File: src/identity-providers/add/DescriptorSettings.tsx (hunk 3 above)
Comment (mfrances17) [reply src/identity-providers/add/DescriptorSettings.tsx]: fixed, always forget this one - thanks

File: src/identity-providers/add/ReqAuthnConstraintsSettings.tsx (hunk 4 above)
Comment (mfrances17) [reply src/identity-providers/add/ReqAuthnConstraintsSettings.tsx]: always forget this one as well - fixed, ty

File: src/identity-providers/add/DescriptorSettings.tsx (hunk 3 above)
Comment (jonkoops) [reply src/identity-providers/add/DescriptorSettings.tsx]: Yeah, might be worth adding a linting rule for it.

**Review by jonkoops — APPROVED** [review]

**Pull request merged**